### PR TITLE
[NCL-7207] Build Config create/edit page - edit callback

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

File: src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx [hunk 1]
```diff
@@ -753,62 +867,64 @@ export const BuildConfigCreateEditPage = ({ isEditPage = false }: IBuildConfigCr
...
+                  configChanges={buildConfigChanges}
+                  onCancel={(buildConfig: BuildConfiguration) => {
+                    cancelBuildConfigChange(buildConfig);
+                  }}
+                />
+              </GridItem>
+            </Grid>
+          </ExpandableSection>
+        </ContentBox>
+      )}
```
Comment (matedo1) [inline src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx:927]: Line 935: wrong submit button naming, see for example `ProjectCreateEditPage.tsx`

File: src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx [hunk 2]
```diff
@@ -253,14 +258,62 @@ export const BuildConfigCreateEditPage = ({ isEditPage = false }: IBuildConfigCr
...
+      buildType: data.buildType,
+      buildScript: data.buildScript,
+      brewPullActive: data.brewPullActive,
+      scmRepository: selectedScmRepository,
+      scmRevision: data.scmRevision,
+      productVersion: selectedProductVersion,
+      parameters: Object.fromEntries(Object.entries(buildParamData).map(([k, v]) => [k, v.value])),
+    } as BuildConfiguration;
+
+    const patchData = createSafePatch(serviceContainerEditPageGet.data!, buildConfig);
```
Comment (matedo1) [inline src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx:274]: There is high amount of unchanged parameters when only Name is changed and submitted.

File: src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx [hunk 3]
```diff
@@ -253,14 +258,62 @@ export const BuildConfigCreateEditPage = ({ isEditPage = false }: IBuildConfigCr
   };
 
   const submitEdit = (data: IFieldValues) => {
```
Comment (matedo1) [inline src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx:260]: Once form is submitted, there is high amount of cancelled requests, the reasons needs to be investigated and new Jira needs to be created if it's a new bug.

**Review by matedo1 — CHANGES_REQUESTED** [review]

**Comment by matedo1** [comment]
Let's continue with another testing round after current bugs are fixed.

File: src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx (hunk 2 above)
Comment (patrikk0123) [reply src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx:274]: Only nested entities' ids are now compared.

File: src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx (hunk 3 above)
Comment (patrikk0123) [reply src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx:260]: The has already been cretaed before: NCL-8349.

**Comment by matedo1** [comment]
I tested it locally and I added jira comments with issues

**Review by matedo1 — APPROVED** [review]

**Pull request merged**